### DISCUSSION TITLE
How DI works on the Endpoints?

### DISCUSSION THREAD

**Discussion by rawgerjmz** [jasontaylordev/CleanArchitecture#1184] [Q&A]
Since it is creating the instances at runtime with ActivatorUtilities, how would you suggest to inject the services (interfaces) on the controllers/Endpoints?

thanks in advance.

**Comment by caner-aras** [comment]
> Since it is creating the instances at runtime with ActivatorUtilities, how would you suggest to inject the services (interfaces) on the controllers/Endpoints?
> 
> thanks in advance.

did you find any solution about that @hamgerjmz?

**Reply by rawgerjmz** [reply]
Yes, I've ended up by retreiving the dependencies from DI containter like this:
![image](https://github.com/jasontaylordev/CleanArchitecture/assets/16430530/b10863ba-6c6d-41fc-9b42-10e722ad4c14)

then you can do var myDependency = scope.ServiceProvider.GetRequiredService<>();
